homeassistant/components/atome/sensor.py

```diff
@@ -0,0 +1,319 @@
+        """Update the daily power usage."""
+        data.get_daily_usage()
+
+    @Throttle(WEEKLY_SCAN_INTERVAL)
+    def update_weekly():
+        """Update the weekly power usage."""
+        data.get_weekly_usage()
+
+    @Throttle(MONTHLY_SCAN_INTERVAL)
+    def update_monthly():
+        """Update the monthly power usage."""
+        data.get_monthly_usage()
+
+    @Throttle(YEARLY_SCAN_INTERVAL)
+    def update_yearly():
+        """Update the yearly power usage."""
+        data.get_yearly_usage()
+
+    update_live()
+    update_daily()
+    update_weekly()
+    update_monthly()
+    update_yearly()
+
+    add_entities([AtomeSensor(data, LIVE_NAME, LIVE_TYPE, update_live)], True)
+    add_entities([AtomeSensor(data, DAILY_NAME, DAILY_TYPE, update_daily)], True)
+    add_entities([AtomeSensor(data, WEEKLY_NAME, WEEKLY_TYPE, update_weekly)], True)
+    add_entities([AtomeSensor(data, MONTHLY_NAME, MONTHLY_TYPE, update_monthly)], True)
+    add_entities([AtomeSensor(data, YEARLY_NAME, YEARLY_TYPE, update_yearly)], True)
+
+
+class AtomeData:
+    """Stores data retrieved from Neurio sensor."""
+
+    def __init__(self, username, password):
+        """Initialize the data."""
+        self.username = username
+        self.password = password
+        self._live_power = None
+        self._subscribed_power = None
+        self._is_connected = None
+        self._daily_usage = None
+        self._daily_price = None
+        self._weekly_usage = None
+        self._weekly_price = None
+        self._monthly_usage = None
+        self._monthly_price = None
+        self._yearly_usage = None
+        self._yearly_price = None
+
+        self._state = None
+
+        try:
+            self.atome_client = AtomeClient(username, password)
+            self.atome_client.login()
+        except PyAtomeError as exp:
+            _LOGGER.error(exp)
+            return
+
+    @property
+    def live_power(self):
+        """Return latest active power value."""
+        return self._live_power
+
+    @property
+    def subscribed_power(self):
+        """Return latest active power value."""
+        return self._subscribed_power
+
+    @property
+    def is_connected(self):
+        """Return latest active power value."""
+        return self._is_connected
+
+    def get_live_power(self):
+        """Return current power value."""
+        try:
+            values = self.atome_client.get_live()
+            self._live_power = values["last"]
+            self._subscribed_power = values["subscribed"]
+            self._is_connected = values["isConnected"]
+            _LOGGER.debug(
+                "Updating Atome live data. Got: %d, isConnected: %s, subscribed: %d",
+                self._live_power,
+                self._is_connected,
+                self._subscribed_power,
+            )
+
+        except KeyError as error:
+            _LOGGER.error("Missing last value in values: %s: %s", values, error)
+            return None
```

Comment:
Remove return.

Probably rename the methods that doesn't return anything from `get_` to `update_`.

Reply:
ok